src/View/BaseViewTrait.php

```diff
@@ -53,13 +53,20 @@ private function getProps(): array
     {
         $props = [];
         $only = $this->getPartialData();
-        $onlyViewVars = ! empty($only) ? $only : array_keys($this->viewVars);
+        $onlyViewVars = !empty($only) ? $only : array_keys($this->viewVars);
         $passedViewVars = $this->viewVars;
 
-        $this->viewVars = [];
+        $nonInertiaProps = $this->getConfig('_nonInertiaProps') ?? [];
+
+        $onlyViewVars = array_diff($onlyViewVars, $nonInertiaProps);
```

Comment:
I would suggest to move these lines of code into a separate (private) method.

For example:
```php
        $onlyViewVars = $this->filterViewVars();
```